### PR TITLE
[WIP] add test checkpoint_01

### PR DESCRIPTION
**Pull request by tjhei**
This test creates a checkpoint and continues the run, then resumes
ASPECT from that checkpoint and check that we get the same result. Setup
is somewhat tricky and is controlled via the plugin in the .cc

### CODE REVIEW

**Comment by tjhei** [comment]
Note: this is work in progress. The solutions are not identical right now.

**Comment by bangerth** [comment]
Should I mark this as `[WIP]`?

**Comment by bangerth** [comment]
Any news?

**Comment by tjhei** [comment]
The status is that the iterative solver is not deterministic (the ML problem we talked about offline).

I now switched to the direct solver, which makes the tests pass. Maybe we should merge it as is for now?

Other ideas:
- Play with the iterative solver options to find a deterministic option.
- Try PETSc+hypre and see what happesn.

File: include/aspect/postprocess/depth_average.h [hunk 1]
```diff
@@ -143,7 +137,7 @@ namespace aspect
          * the current time to avoid falling behind with next_output_time and
          * having to catch up once the time step becomes larger.
          */
-        void set_next_output_time (const double current_time);
+        void set_last_output_time (const double current_time);
```
Comment (bangerth) [inline include/aspect/postprocess/depth_average.h]: I think you need to update the documentation as well.

File: tests/checkpoint_01.cc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  int ret;
+  
+  ret = system ("cd output-checkpoint_01 ; "
+	  "(cat " ASPECT_SOURCE_DIR "/tests/checkpoint_01.prm "
+	  " ; "
+	  " echo 'set Output directory = output1.tmp' "
+	  " ; "
+	  " rm -rf output1.tmp ; mkdir output1.tmp "
+	  ") "
+	  "| ../../aspect -- >/dev/null ");
```
Comment (bangerth) [inline tests/checkpoint_01.cc]: Wow, that's a mouthful :-)

**Comment by bangerth** [comment]
Looks good. Fix the documentation of the one function and merge, assuming the tests complete successfully.

As for the other suggestions -- I'm for them, but these can be separate PRs and tests then. I think the test as it is right now already serves a good purpose.

**Pull request merged**